plugins/chat/spec/system/mention_warnings_spec.rb

```diff
@@ -69,5 +70,42 @@
         end
       end
     end
+
+    context "when channel has allow_channel_wide_mentions disabled" do
+      before { channel_1.update(allow_channel_wide_mentions: false) }
+
+      %w[@here @all].each do |mention_text|
+        it "displays a warning" do
+          chat_page.visit_channel(channel_1)
+          chat_channel_page.type_in_composer(mention_text)
+
+          expect(page).to have_css(".chat-mention-warnings")
+          expect(page.find(".chat-mention-warnings-list__simple")).to be_present
+        end
+      end
+
+      it "Retains warnings when loading drafts or changing channels with no draft" do
+        Chat::Draft.create!(
+          chat_channel: channel_1,
+          user: current_user,
+          data: { message: "@all" }.to_json,
+        )
+        chat_page.visit_channel(channel_1)
+
+        # Channel 1 has a draft that causes a mention warning. Should appear on load
+        expect(page).to have_css(".chat-mention-warnings")
+        expect(page.find(".chat-mention-warnings-list__simple")).to be_present
+
+        # Channel 2 doesn't have a draft so it should disappear
+        chat_page.visit_channel(channel_2)
+        expect(page).not_to have_css(".chat-mention-warnings")
```

Comment:
```suggestion
        expect(page).to have_no_css(".chat-mention-warnings")
```

Reply:
This will be slow otherwise, as we will wait full duration to ensure it's not present